### PR TITLE
Add renovate for dependency updates

### PR DESCRIPTION
**Pull request by timebertt**
**What this PR does / why we need it**:

This PR adds renovate to this repository so that we can easily keep dependencies up-to-date.

**Which issue(s) this PR fixes**:
n/a

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      ]
+    }
+  ],
+  "packageRules": [
+    {
+      // automerge non-major updates except 0.* versions
+      // similar to :automergeStableNonMajor preset, but also works for versioning schemes without range support
+      "matchUpdateTypes": ["minor", "patch"],
+      "matchCurrentVersion": "!/^v?0\\./",
+      "automerge": true
```
Comment (dergeberl) [inline .github/renovate.json5:30]: Do we really want minor patches automerge. Some versions we excluded down below but I would think automerge minor patches could lead to issues (even if it should not when everyone is following the semver)

Comment (timebertt) [reply .github/renovate.json5]: Fine for me, let's start without automerging minor updates.

File: .github/renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        // added packages
+        "sigs.k8s.io/controller-runtime",
+        "github.com/gardener/gardener"
+      ]
+    },
+    {
+      // automerge k8s.io/utils updates
+      "matchDatasources": ["go"],
+      "matchPackageNames": ["k8s.io/utils"],
+      "matchUpdateTypes": ["digest"],
```
Comment (robinschneider) [inline .github/renovate.json5:72]: Why is this needed and how does this work?

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -1,35 +1,22 @@
...
-      with:
-        path: go/src/github.com/stackitcloud/gardener-extension-acl
     - uses: actions/setup-go@v4
       with:
-        go-version-file: go/src/github.com/stackitcloud/gardener-extension-acl/go.mod
-        cache-dependency-path: go/src/github.com/stackitcloud/gardener-extension-acl/go.sum
+        go-version-file: go.mod
     - name: Configure git
       run: |
         git config --global user.email "ci@gardener.cloud"
```
Comment (robinschneider) [inline .github/workflows/verify.yml:22]: Is this intended to be a `@gardener.cloud` mail domain?

File: .github/renovate.json5 (hunk 2 above)
Comment (timebertt) [reply .github/renovate.json5:72]: I'm not sure if I get your question. The package is always referenced by digest and this is how you tell renovate to automerge updates: https://github.com/stackitcloud/gardener-extension-acl/blob/981945f35dc19535e3d4e97ed9311c326b8ca75b/go.mod#L31

Comment (robinschneider) [reply .github/renovate.json5:72]: I was wondering, how does renovate know if a package is newer, if it uses a digest instead of semver. But based on this example it could be alphabetical.

File: .github/workflows/verify.yml (hunk 3 above)
Comment (timebertt) [reply .github/workflows/verify.yml:22]: This is just a dummy user that was once needed for gardener's `check-generate.sh` script.
I dropped this step from the workflow and replaced gardener's check-generate with dedicated make targets that are much simpler and don't write any commits.

**Review by robinschneider — APPROVED** [review]
